api/admin/controller.py

```diff
@@ -1061,114 +1061,129 @@ class CustomListsController(CirculationManagerController):
     def custom_lists(self):
         library = flask.request.library
 
-        data_source = DataSource.lookup(self._db, DataSource.LIBRARY_STAFF)
-
         if flask.request.method == "GET":
             custom_lists = []
             for list in library.custom_lists:
-                entries = []
-                for entry in list.entries:
-                    if entry.edition:
-                        entries.append(dict(pwid=entry.edition.permanent_work_id,
-                                            title=entry.edition.title,
-                                            authors=[author.display_name for author in entry.edition.author_contributors],
-                        ))
                 collections = []
                 for collection in list.collections:
                     collections.append(dict(id=collection.id, name=collection.name, protocol=collection.protocol))
-                custom_lists.append(dict(id=list.id, name=list.name, entries=entries, collections=collections))
+                custom_lists.append(dict(id=list.id, name=list.name, collections=collections, entry_count=len(list.entries)))
```

Comment:
Obviously the big change is the one you made, but we could get all of this stuff -- custom lists for a library with counts and collections for each list -- with a single database query if necessary.

Reply:
It seems fast enough already, but I'll come back to it later if it's a problem. Possibly if you have a large number of lists it would be important.